turbopack/crates/turbo-tasks-fs/src/lib.rs

```diff
@@ -1180,57 +1180,54 @@ impl FileSystemPath {
     /// contain ".." or "." seqments, but it must not leave the root of the
     /// filesystem.
     #[turbo_tasks::function]
-    pub async fn join(self: Vc<Self>, path: RcStr) -> Result<Vc<Self>> {
-        let this = self.await?;
-        if let Some(path) = join_path(&this.path, &path) {
-            Ok(Self::new_normalized(*this.fs, path.into()))
+    pub fn join(&self, path: RcStr) -> Result<Vc<Self>> {
```

Comment:
dropping `async` also tags this as `immutable`

is that correct? since it needs to read `Vc<Self>`` to be executed?

This is kind of unrelated to your PR... sorry

Reply:
i don't think we have a ton of test coverage for invalidations